cli/running/watchdog.go

```diff
@@ -28,13 +30,22 @@ type Watchdog struct {
 	// done channel used to inform the signal handle goroutine
 	// about termination of the Instance.
 	done chan bool
+	// createInstance is a function used for instance creation.
+	createInstance func(cmdCtx *cmdcontext.CmdCtx, args []string) (*Instance, error)
+	// args is arguments passed to the programm.
+	args []string
+	// CmdCtx is the main structure of the program context.
+	// Contains within itself other structures of CLI modules.
+	cmdCtx *cmdcontext.CmdCtx
 }
 
 // NewWatchdog creates a new instance of Watchdog.
-func NewWatchdog(instance *Instance, restartable bool,
-	restartTimeout time.Duration, logger *ttlog.Logger) *Watchdog {
-	wd := Watchdog{Instance: instance, logger: logger, restartable: restartable,
-		restartTimeout: restartTimeout}
+func NewWatchdog(restartable bool,
+	restartTimeout time.Duration, logger *ttlog.Logger,
+	createInstance func(cmdCtx *cmdcontext.CmdCtx, args []string) (*Instance, error),
```

Comment:
Didn't you consider using type alias for this function to do not repeat type declaration?